### PR TITLE
Add AdaBelief optimizer

### LINKED ISSUE #2203: Add AdaBelief optimizer [https://arxiv.org/abs/2010.07468]
**Describe the feature and the current behavior/state.**

**Relevant information**
- Are you willing to contribute it (yes/no): no
- Are you willing to maintain it going forward? (yes/no): no 
- Is there a relevant academic paper?  https://arxiv.org/abs/2010.07468
- Is there already an implementation in another framework? https://github.com/juntang-zhuang/Adabelief-Optimizer
- Was it part of tf.contrib? No

**Which API type would this fall under (layer, metric, optimizer, etc.): Optimizer/Layer

**Who will benefit with this feature?** Everybody; shows better results/more promising ones than the other optimizers

**Any other info.**

### PR DESCRIPTION
**Pull request by cryu854**
# Description

Fixes #2203 by adding AdaBelief optimizer. [Project page](https://juntang-zhuang.github.io/adabelief/), [arXiv](https://arxiv.org/abs/2010.07468), [GitHub](https://github.com/juntang-zhuang/Adabelief-Optimizer)

Although @juntang-zhuang has an official implementation at [juntang-zhuang/Adabelief-Optimizer](https://github.com/juntang-zhuang/Adabelief-Optimizer), but it was heavily based on [tensorflow_addons/optimizers/rectified_adam.py](https://github.com/tensorflow/addons/blob/v0.11.2/tensorflow_addons/optimizers/rectified_adam.py#L24-L326), and we only modified the core algorithm part. 

So we decided to contribute this optimizer to leverage the friendly API from addons, with the hope that it can help anyone who is struggling for training a neural network.

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [x] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [x] This PR addresses an already submitted #2203  issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

Unit tests for AdaBelief, the results are compared with numpy update using the same algorithm.

### CODE REVIEW

**Comment by bhack** [comment]
I think It is a little bit early for this. See https://github.com/tensorflow/addons/issues/2194

**Comment by cryu854** [comment]
Thank you for your feedback. I think AdaBelief would be a good choice for training because it uses curvature information like an ideal optimizer, which makes it more intuitively reasonable. Once addons decides to add this feature, I will help maintain it.

**Comment by bhack** [comment]
Is that currently we have just a single citation https://github.com/tensorflow/addons/issues/2203#issuecomment-724610801

**Comment by dathudeptrai** [comment]
@bhack @cryu854 will it merge ?

**Comment by cryu854** [comment]
@dathudeptrai I think it depends on the usage rate by community and the number of citations. Currently, please use [juntang-zhuang/Adabelief-Optimizer](https://github.com/juntang-zhuang/Adabelief-Optimizer).

**Comment by denadai2** [comment]
isn't this done?

**Comment by bhack** [comment]
Yes
